### ISSUE TITLE
VoiceChat demo is broken in Unity 5

### ISSUE THREAD

**Issue by erebuswolf** [fholm/unityassets#2]
I tried forking this repo and getting the voicechat demos to work in the newest version of Unity. They run after the program auto upgrades them, there is a minor warning about a deprecated function call that doesn't matter. But the microphone never seems to pick up audio. I can run a server and two clients locally and the audio never gets from one client to the other (because it is never picked up). Also, when I turn auto detect on it never picks it up. I assume this is a problem with new unity and microphone apis or something. I tried it on windows 8 and Mac. It isn't the microphone. I'm debugging this separately but figured I'd post here in case you wanted to give it a look. Really hoping to get your code working it looks great for what I need.

**Comment by mat27** [comment]
Fholm doesn't provide any support for this I'm pretty sure. As its his old
projects. Go with something like uvoice or dfvoice for out of the box
functionality

**Comment by kgreenwoodSHAI** [comment]
@erebuswolf, I also tried running the voicechat demo and had the same problem as you did - did you end up digging into the compatibility issue further?

**Comment by erebuswolf** [comment]
Yes, though it wasn't a bug with the program I was just an idiot. Change
the IP address, that is a public variable (settable through the UI) to
127.0.0.1 (local loopback address). I forget which object had the IP
address but if you do that it will work out of box. The default value is
whatever the original author's local network IP was when he wrote it. I
made a fork of the repo and made a push request with an updated version
that also makes the debug feature work different but the original dev
either doesn't want it or didn't see it. If that doesn't work I suggest
making a project that just creates a mic listener object and plays back
whatever the mic record on a 5 second delay to confirm it isn't a mic
compatibility issue.

On Wed, May 27, 2015 at 2:49 PM, kgreenwoodSHAI <notifications@github.com>
wrote:

> @erebuswolf <https://github.com/erebuswolf>, I also tried running the
> voicechat demo and had the same problem as you did - did you end up digging
> into the compatibility issue further?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fholm/unityassets/issues/2#issuecomment-106088369>.
>



-- 
Jesse

**Comment by kgreenwoodSHAI** [comment]
Right, so I've been working with 127.0.0.1, but still haven't been able to get it to work (but I'm also super new to Unity, so it's very possible that I'm doing something else stupid), though I'll definitely try your suggestion with the mic listener object.  So, you were able to get the demo to work - what about the rest of the package?  Have you successfully been able to integrate it into your own project?

**Comment by erebuswolf** [comment]
I was able to build it and launch multiple instances of the project and it
worked fine. Just built a solution and opened the output executable and ran
the project locally in unity for debugging at the same time. I have not
made steps to integrate it yet, been working on other aspects of the game.

On Wed, May 27, 2015 at 2:59 PM, kgreenwoodSHAI <notifications@github.com>
wrote:

> Right, so I've been working with 127.0.0.1, but still haven't been able to
> get it to work (but I'm also super new to Unity, so it's very possible that
> I'm doing something else stupid), though I'll definitely try your
> suggestion with the mic listener object. So, you were able to get the demo
> to work - what about the rest of the package? Have you successfully been
> able to integrate it into your own project?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fholm/unityassets/issues/2#issuecomment-106090535>.
>



-- 
Jesse

**Comment by kgreenwoodSHAI** [comment]
Yep - totally just got it to work - when I was fiddling with things, I had changed the port number, but when I changed it back to 15000, it worked great!  Thanks for all your help!

**Comment by erebuswolf** [comment]
Glad to hear you got it working! Good luck with your game!

On Wed, May 27, 2015 at 3:10 PM, kgreenwoodSHAI <notifications@github.com>
wrote:

> Yep - totally just got it to work - when I was fiddling with things, I had
> changed the port number, but when I changed it back to 15000, it worked
> great! Thanks for all your help!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fholm/unityassets/issues/2#issuecomment-106093902>.
>



-- 
Jesse

**Comment by Jalict** [comment]
I found that @garrafote have made a new version which works with Unity3D 5. In case people are interested, check out this repo https://github.com/garrafote/Unity-Voice-Chat/

**Comment by garrafote** [comment]
My bad! I totally forgot that the VoiceChat files are originally from this repository. I will submit a new pull request with my most recent version and deprecate this other repository that I created.

**Comment by srirachaQueen** [comment]
Has anyone been able to get the Auto-detect to work?  Even with a variety of Autodetect index values, it's transmitting without differentiation whether someone is speaking or not.

**Comment by NoahMilam** [comment]
I need a little help getting started with your sample project Networking [HLAPI].
I am struggling with this , var proxy = Instantiate<GameObject>(proxyPrefab);
ArgumentException: this thing you want to instance is null.
this happens when I build and try to connect a client to the server. A am using your voiceChat_player with a network ID added to it. any help would be great. thanks

**Comment by asd2846531** [comment]
Has anyone encountered the "Network Id not set" problem mentioned in this thread?http://forum.unity3d.com/threads/voice-chat-for-unity-networking.400597/

**Comment by NoahMilam** [comment]
that is the exact problem I am having. when turn off local debug I get it. Im going through it today and ill let you know if I find anything.

**Comment by paganini24** [comment]
@NoahMilam you need to run one instance server only, if you run host, your server id will be 0 always on server. You have to connect as client with your players

**Comment by byrd1bar** [comment]
Hi,

I'm fairly new to Unity development (~6 months of development).  I got the latest VoiceChat package from github and got the local debug voice playback to work verifying that my Oculus VR headphones and microphone work.  I basically encountered the same "Network Id =0" problem.  I get this problem even when I connect as a client with one instance of the server.  I added the Network Identity component to the VoiceChat_NetworkProxy and during debug, the netID from the Network Identity component is non-zero while the Network Id on the same VoiceChat_NetworkProxy is still zero!  I have multiplayer support using Unity's NetworkManager and I created the VoiceChat game object using the steps from the VoiceChat pdf doc.

Does anyone have working sample code showing how to use VoiceChat with Unity 5.3 or later?  In particular, that shows how to create the VoiceChat_NetworkProxy using the latest NetworkServer.Spawn() method instead of the old Network.Instantiate()?  And also show where to call this method?

Thanks
